tensorflow/core/kernels/data/batch_dataset_op_test.cc

```diff
@@ -11,194 +11,119 @@ limitations under the License.
- public:
-  Status Initialize(BatchDatasetParams* batch_dataset_params) override {
-    TF_RETURN_IF_ERROR(InitThreadPool(thread_num_));
-    TF_RETURN_IF_ERROR(InitFunctionLibraryRuntime({}, cpu_num_));
-
-    // Populate the `input_dataset` in `batch_dataset_params_`.
-    RangeDatasetParams input_dataset_params(
-        0, batch_dataset_params->num_input_elements, 1, {DT_INT64},
-        {PartialTensorShape({})}, "range_dataset");
-    TF_RETURN_IF_ERROR(MakeRangeDataset(input_dataset_params,
-                                        &batch_dataset_params->input_dataset));
-    // Create the dataset kernel.
-    TF_RETURN_IF_ERROR(
-        MakeDatasetOpKernel(*batch_dataset_params, &dataset_kernel_));
-    // Create the inputs for the dataset op.
-    gtl::InlinedVector<TensorValue, 4> inputs;
-    TF_RETURN_IF_ERROR(batch_dataset_params->MakeInputs(&inputs));
-    // Creat the dataset context.
-    TF_RETURN_IF_ERROR(
-        CreateDatasetContext(dataset_kernel_.get(), &inputs, &dataset_ctx_));
-    // Create the dataset.
-    TF_RETURN_IF_ERROR(
-        CreateDataset(dataset_kernel_.get(), dataset_ctx_.get(), &dataset_));
-
-    // Create the iterator context.
-    TF_RETURN_IF_ERROR(
-        CreateIteratorContext(dataset_ctx_.get(), &iterator_ctx_));
-    // Create the iterator.
-    TF_RETURN_IF_ERROR(dataset_->MakeIterator(
-        iterator_ctx_.get(), batch_dataset_params->iterator_prefix,
-        &iterator_));
-    return Status::OK();
-  }
-
- protected:
-  // Creates a new `BatchDataset` op kernel.
-  Status MakeDatasetOpKernel(
-      const BatchDatasetParams& dataset_params,
-      std::unique_ptr<OpKernel>* batch_dataset_op_kernel) override {
-    name_utils::OpNameParams params;
-    params.op_version = kOpVersion;
-    NodeDef node_def = test::function::NDef(
-        dataset_params.node_name,
-        name_utils::OpName(BatchDatasetOp::kDatasetType, params),
-        {BatchDatasetOp::kInputDataset, BatchDatasetOp::kBatchSize,
-         BatchDatasetOp::kDropRemainder},
-        {{BatchDatasetOp::kParallelCopy, dataset_params.parallel_copy},
-         {BatchDatasetOp::kOutputTypes, dataset_params.output_dtypes},
-         {BatchDatasetOp::kOutputShapes, dataset_params.output_shapes}});
-    TF_RETURN_IF_ERROR(CreateOpKernel(node_def, batch_dataset_op_kernel));
-    return Status::OK();
-  }
-};
+class BatchDatasetOpTest : public DatasetOpsTestBaseV2 {};
 
 // Test Case 1: test BatchDatasetV2 with `drop_remainder` = false and a batch
 // size that can evenly split the input dataset.
 BatchDatasetParams BatchDatasetParams1() {
-  return {/*num_input_elements=*/12,
-          /*batch_size=*/4,
-          /*drop_remainder=*/false,
-          /*parallel_copy=*/true,
-          /*output_dtypes=*/{DT_INT64},
-          /*output_shapes=*/{PartialTensorShape({4})},
-          /*node_name=*/kNodeName};
+  auto range_dataset_params = RangeDatasetParams(0, 12, 1);
+  return BatchDatasetParams(std::move(range_dataset_params),
```

Comment:
```suggestion
  return BatchDatasetParams(RangeDatasetParams(0, 12, 1),
```

Reply:
If my understanding is correct, `std::move(range_dataset_params)` is used to avoid a copy here.

Reply:
I believe the compiler will know to elide the copy for the temporary RangeDatasetParams object in this case -- https://en.cppreference.com/w/cpp/language/copy_elision

Reply:
Yes, you are right. Will change it.